solutions/ioi-11-race.mdx

```diff
@@ -0,0 +1,84 @@
+---
+id: ioi-11-race
+source: IOI 2011
+title: Race
+author: Timothy Gao
+---
+
+[Official Analysis with Centroid Decomposition Downloaded Here](https://ioinformatics.org/page/ioi-2011/37)
+
+## Alternative Solution: Small to Large Merging
+
+Like many centroid decomposition problems, race is also solvable via the small-to-large merging technique. At each node, consider storing a map. Naively, storing pairs of Length : #_of_paths does not work, because when we move from a node to it's ancestor, all values in the map stored that node much be updated. To account for this, we can instead define all paths as global to the root. In other words, we store pairs of Length_from_root : height. This way, when we merge a smaller map to a larger one at node $u$, we can process the answer with the formulation 
+
+$\textit{min(answer, smallmapheight + largemap[K-smallmaplength] - 2*height[u])}$
+
+
+## Implementation
+
+**Time Complexity:** $\mathcal{O}(N\log^2N)$
+
+
+```cpp
+#include <bits/stdc++.h>
+ 
```

Comment:
right now it looks like they're spaces and not tabs you should use the editor on the guide website to change this it will appear as an arrow instead of dots when you command A after the change